tests/unittests/BackendTestUtils.h

```diff
@@ -113,6 +113,15 @@ static const auto all_backends = ::testing::Values(
   if (!isEnabledBackend({__VA_ARGS__}))                                        \
     return;
 
+/// Blacklist of tests for the current backend under test.
+extern std::set<std::string> backendTestBlacklist;
+
+/// Helper macro to check the current test against the blacklist.
+#define CHECK_IF_ENABLED()                                                     \
+  if (backendTestBlacklist.count(                                              \
```

Comment:
Should we have whitelist and blacklist? Depending on the number of elements in each, it could be easier to define one or the other. The check would look like this:
```c++
/// Bail if there is a non-empty white list and the current test is not in it.
if (!backendTestWhitelist.empty() && !backendTestWhitelist.count(...)) {
  return;
}
/// Bail if the current test is in the blacklist.
if (backendTestBlacklist.count(...)) {
   return;
}
```

BTW, we have a similar logic in the partitioner.

Reply:
I wasn't sure about this -- if backends define their support differently then it makes things a bit more error prone. E.g. if I add a new test that I want to enable on the CPU and Habana backends, and I initially test it just on the CPU backend (which may use a blacklist and so it's enabled by default), and then once it's working I forget that I need to explicitly enable it on Habana which use a whitelist.

I'm also wondering about the potential for mistakes if we e.g. misspell names in the whitelist. Then it would appear the test passes even if it's not enabled, as we don't have `GTEST_SKIP()` yet.